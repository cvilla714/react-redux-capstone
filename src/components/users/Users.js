/* eslint-disable react/destructuring-assignment */
/* eslint-disable no-console */
/* eslint-disable react/forbid-prop-types */
/* eslint-disable react/prop-types */

import React from 'react';
// import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import UserItem from './UserItem';
import Spinner from '../layout/Spinner';

const Users = (props) => {
  if (props.loading || props.users === null) {
    return <Spinner />;
  }

  console.log(props);
  const userStyle = {
    display: 'grid',
    gridTemplateColumns: 'repeat(3,1fr)',
    gridGap: '1rem',
  };

  return (
    <div style={userStyle}>
      {props.users.map((user) => (
        <UserItem key={user.id} user={user} />
      ))}
    </div>
  );
};

// Users.propTypes = {
//   users: PropTypes.array.isRequired,
//   loading: PropTypes.bool.isRequired,
// };
const mapStateToProps = (state) => ({
  users: state,
});
// export default Users;
export default connect(mapStateToProps)(Users);
