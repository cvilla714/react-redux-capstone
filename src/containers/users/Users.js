import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import UserItem from '../../components/users/UserItem';
import Spinner from '../../components/layout/Spinner';

const Users = ({ users, loading, filter }) => {
  if (loading) {
    return <Spinner />;
  }

  const userStyle = {
    display: 'grid',
    gridTemplateColumns: 'repeat(3,1fr)',
    gridGap: '1rem',
  };

  const filteringUsers = () => {
    if (filter) {
      return users.filter((user) => user.login.toLowerCase().includes(filter.toLowerCase()));
    }
    return users;
  };

  return (
    <div style={userStyle}>
      {filteringUsers().map((user) => (
        <UserItem key={user.id} user={user} />
      ))}
    </div>
  );
};

Users.defaultProps = {
  filter: '',
};

Users.propTypes = {
  users: PropTypes.arrayOf(Array).isRequired,
  loading: PropTypes.bool.isRequired,
  filter: PropTypes.string,
};

const mapStateToProps = (state) => ({
  users: state.github.users,
  loading: state.github.loading,
  filter: state.filter,
});

export default connect(mapStateToProps)(Users);
