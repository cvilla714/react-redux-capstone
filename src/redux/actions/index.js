import axios from 'axios';
import {
  CLEAR_USERS,
  GET_REPOS,
  GET_USER,
  SEARCH_USERS,
  SET_LOADING,
  LOGS_ERROR,
} from './types';

export const setLoading = () => ({
  type: SET_LOADING,
});

export const searchUsers = (text) => async (dispatch) => {
  try {
    setLoading();

    const res = await axios.get(
      `https://api.github.com/search/users?q=${text}`,
      {
        headers: {
          Authorization: `token ${process.env.REACT_APP_GITHUB_TOKEN}`,
        },
      },
    );
    dispatch({
      type: SEARCH_USERS,
      payload: res.data.items,
    });
  } catch (error) {
    dispatch({
      type: LOGS_ERROR,
      payload: error.respose.statusText,
    });
  }
};

export const getUser = (username) => async (dispatch) => {
  try {
    setLoading();

    const res = await axios.get(`https://api.github.com/users/${username}`, {
      headers: {
        Authorization: `token ${process.env.REACT_APP_GITHUB_TOKEN}`,
      },
    });

    dispatch({
      type: GET_USER,

      payload: res.data,
    });
  } catch (error) {
    dispatch({
      type: LOGS_ERROR,
      payload: error.respose.statusText,
    });
  }
};

export const getUserRepos = (username) => async (dispatch) => {
  try {
    setLoading();

    const res = await axios.get(
      `https://api.github.com/users/${username}/repos?per_page=5&sort=created:asc`,
      {
        headers: {
          Authorization: `token ${process.env.REACT_APP_GITHUB_TOKEN}`,
        },
      },
    );

    dispatch({
      type: GET_REPOS,
      payload: res.data,
    });
  } catch (error) {
    dispatch({
      type: LOGS_ERROR,
      payload: error.respose.statusText,
    });
  }
};

export const clearUsers = () => ({
  type: CLEAR_USERS,
});
