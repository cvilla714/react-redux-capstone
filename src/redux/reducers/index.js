import { combineReducers } from 'redux';
import githubReducer from './githubReducer';
import alertReducer from './alertReducer';
import filterReducer from './filterReducer';

export default combineReducers({
  github: githubReducer,
  alert: alertReducer,
  filter: filterReducer,
});
