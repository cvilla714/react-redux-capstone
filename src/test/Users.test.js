import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { createStore, applyMiddleware } from 'redux';
import combineReducers from '../reducers';
import Users from '../containers/users/Users';

test('Users renders correctly', () => {
  const store = createStore(combineReducers, applyMiddleware(thunk));
  const tree = create(
    <Provider store={store}>
      <Users />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
