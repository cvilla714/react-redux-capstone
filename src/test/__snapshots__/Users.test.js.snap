// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users renders correctly 1`] = `
<div
  style={
    Object {
      "display": "grid",
      "gridGap": "1rem",
      "gridTemplateColumns": "repeat(3,1fr)",
    }
  }
/>
`;
